File: frontend/src/app/types/annotation.ts
export interface Annotation {
  id: number;
  kind: 'manual' | 'alert' | 'maintenance' | 'event';
  scope_kind: 'sensor' | 'asset' | 'dashboard' | 'global';
  scope_id: number | null;
  label: string;
  body: string | null;
  started_at: string;
  ended_at: string | null;
  color: string | null;
  source_event_id: number | null;
  created_by: number | null;
  created_at: string;
}

File: backend/DAL/models/annotation.py
from datetime import datetime

from sqlalchemy import BigInteger, DateTime, ForeignKey, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from DAL.db_context import Base
from DAL.models._mixins import CreatedAtMixinTZ


class Annotation(CreatedAtMixinTZ, Base):
    __tablename__ = "annotation"

    id: Mapped[int] = mapped_column(Integer, primary_key=True)
    kind: Mapped[str] = mapped_column(String(20), nullable=False)
    scope_kind: Mapped[str] = mapped_column(String(20), nullable=False)
    scope_id: Mapped[int | None] = mapped_column(Integer, nullable=True)
    label: Mapped[str] = mapped_column(String(160), nullable=False)
    body: Mapped[str | None] = mapped_column(Text, nullable=True)
    started_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), nullable=False)
    ended_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)
    color: Mapped[str | None] = mapped_column(String(20), nullable=True)
    source_event_id: Mapped[int | None] = mapped_column(
        BigInteger, ForeignKey("alert_event.id", ondelete="SET NULL"), nullable=True
    )
    created_by: Mapped[int | None] = mapped_column(
        Integer, ForeignKey("users.id", ondelete="SET NULL"), nullable=True
    )
